plugins/CoreHome/vue/src/PeriodSelector/PeriodSelector.vue

```diff
@@ -259,14 +262,18 @@ export default defineComponent({
       window.$(this.$refs.root as HTMLElement).parent('#periodString').show();
     });
 
-    this.updateSelectedValuesFromHash();
-    watch(() => MatomoUrl.parsed.value, this.updateSelectedValuesFromHash);
-
     this.isComparing = ComparisonsStore.isComparingPeriods();
     watch(() => ComparisonsStore.isComparingPeriods(), (newVal) => {
       this.isComparing = newVal;
     });
 
+    this.updateSelectedValuesFromHash();
+    this.updateComparisonValuesFromHash();
+    watch(() => MatomoUrl.parsed.value, () => {
```

Comment:
Just a question, does this watch mean we support changing the URL without page reload, if that's the case have you covered this behaviour in your tests?

Reply:
With the changed watch it is now perhaps a bit less obvious, but yes, any hash change was/is directly resulting in a change to the selector content. The changed tests perhaps show that a bit better now as the `about:blank` step was removed.